lib/hash.rb

```diff
@@ -1,12 +1,14 @@
 class Hash
   class << self
     attr_accessor :use_dot_syntax
+    attr_accessor :hash_dot_use_default
   end
 
   attr_accessor :use_dot_syntax
+  attr_accessor :hash_dot_use_default
 
-  def to_dot
-    dotify_hash(self)
+  def to_dot(use_default=false)
```

Comment:
Could we convert this to a named argument? I think this will be clearer for the user. e.g.

```
{}.to_dot(use_default: true)
```

Reply:
Sure - I had assumed the default would be `false` to preserve current behavior, but do you want me to make it `true`?  That would be a breaking change for anyone relying on current behavior.  I think its a better default, but I only really know my use case.

Reply:
Agreed. Sorry for the confusing example.